q2_feature_table/_conditional_filter.py

```diff
@@ -0,0 +1,33 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+
+import biom
+
+
+def filter_features_conditionally(table: biom.Table,
+                                  prevelance: float, abundance: float
+                                  ) -> biom.Table:
+    """
+    A function to perform joint filtering because it makes life better
+    """
+    num_observations, num_samples = table.shape
+    prevelance = prevelance * num_samples
+
+    # Calculates the filteering parameters on the original table
+    def _filter_f(v, id_, md):
+        return (v > abundance).sum() >= prevelance
```

Comment:
Same suggestion as below:  since we have plenty of space here, I suggest more verbose naming, e.g., `values`, and `metadata`.

Reply:
This is the syntax used in the [biom documentation](https://biom-format.org/documentation/generated/biom.table.Table.filter.html#biom.table.Table.filter). When given the choice, I like to be consistent with source anonymous function documentation, I think it makes it easier for people working between this code, the biom example code, and their own code to follow what, where, and how the arguments translate in the different cases

Reply:
That's a really good point.  It still seems somewhat ambiguous to me though since the docstring for `ids_to_keep` uses the verbose var names `values`, and `metadata` and the examples don't use the `v` shorthand (they use `val`).  I'm not really sure what the best practice is here (if there is one), to be honest.  My inclination is to prioritize readability and ergonomics in our own codebase.  I'm not sure how much we gain by staying consistent with biom's examples; and those examples could change.  I'm happy to defer to your judgment here since you have more experience.  Maybe we can have the best of both worlds with good documentation. 🤔 

Another nitpick:  `md` to `_` since it isn't actually used. But again, we can make an exception in this case if there is value in keeping the function signature consistent with the biom doc examples.

Reply:
I can switch to the more verbose documentation, and I'd strongly like to keep the metadata argument. I (and I assume other people) learn a lot about the way code is put together by reading codebases. I know pep8/flake8 is supposed to make things more elegant. And I think there's something useful about knowing `biom.filter` requires a 3 argument function (`f(val, id_, md)`) and collapse is a two argument function (`f(id_, md)`) by seeing it in place?